File: library-domain/src/main/java/com/dodecaedro/library/domain/repository/BookRepository.java
package com.dodecaedro.library.domain.repository;

import com.dodecaedro.library.domain.model.Book;

import java.util.List;
import java.util.Optional;
import java.util.UUID;

public interface BookRepository {
	Optional<Book> findById(UUID bookId);

	Book save(Book book);

	Optional<Book> findByIsbn(String isbn);

	List<Book> findAll();

	List<Book> findAllById(Iterable<UUID> ids);

	void delete(Book book);
}

File: library-domain/src/main/kotlin/com/dodecaedro/library/domain/repository/BookRepository.kt
package com.dodecaedro.library.domain.repository

import com.dodecaedro.library.domain.model.Book
import java.util.*

interface BookRepository {

	fun findBookById(bookId: UUID): Book?

	fun save(book: Book): Book

	fun findByIsbn(isbn: String): Book?

	fun findAll(): List<Book>

	fun delete(book: Book)
}
